localstack/services/firehose/provider.py

```diff
@@ -621,9 +620,7 @@ def _put_to_search_db(
                 )
             )
             try:
-                db_connection.create(
-                    index=search_db_index, doc_type=search_db_type, id=obj_id, body=body
-                )
+                db_connection.create(index=search_db_index, id=obj_id, body=body)
```

Comment:
q: Are you sure this is still compatible with older versions of OpenSearch?
According to the [compatibility docs you mentioned, Firehose does not yet support OpenSearch 2.3](https://docs.aws.amazon.com/opensearch-service/latest/developerguide/version-migration.html): 
> Kinesis Data Firehose currently doesn't support domains running version 2.3 as a destination. If you need to receive data from Kinesis Data Firehose, we recommend waiting until it's supported before you upgrade your domains to 2.3.

If this is not compatible, then I would propose the following steps:
- When determining the endpoint, also check if the version is compatible, and raise an error if it's not.
- Update the tests such that they test the error.
- Update the tests such that they use an older / compatible version of OpenSearch for the Firehose integration testing.